src/components/heatmap/heatmap.tsx

```diff
@@ -56,9 +59,13 @@ const Heatmap = ({
   const [invalidConfigs, setInvalidConfigs] = useState<string[]>(['']);
   const [configErrorsPresent, setConfigErrorsPresent] = useState<boolean>(false);
   const [heatmaps, setHeatmaps] = useState<any[]>([]);
+  const [noData, setNoData] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
```

Comment:
Initialize `isLoading` to `true` since the app is "loading" initially until we know all the requests have returned.